lib/rbnacl/hash/blake2b.rb

```diff
@@ -64,19 +76,61 @@ def initialize(opts = {})
 
         @salt     = opts.fetch(:salt, EMPTY_SALT)
         @salt     = Util.zero_pad(SALTBYTES, @salt)
+        @incycle  = false
+        @instate  = nil
       end
 
       # Calculate a Blake2b digest
       #
       # @param [String] message Message to be hashed
       #
       # @return [String] Blake2b digest of the string as raw bytes
-      def digest(message)
+      def digest(*message)
+        return final_digest if message.empty?
```

Comment:
Sure, I taken out the current digest into class method, I also extract the validation from `initialize`, since I don't think I want to instantiate the object and add `attr_reader` just to read the options back.   Is this looks ok for you? I will rebase these commits later if needed